app/src/main/java/com/amaze/filemanager/adapters/data/IconDataParcelable.java

```diff
@@ -64,6 +78,7 @@ public IconDataParcelable(Parcel im) {
         image = im.readInt();
         loadingImage = im.readInt();
         isImageBroken = im.readInt() == 1;
+        hashCode = im.readInt();
```

Comment:
What if `hashCode` was null?

Reply:
What are you reading in `im.readInt()` then?

Reply:
It'll reference hashCode  to null in case hashCode was null. We don't save hashcode for general file types, but only for cloud storages. So in case of cloud storage if hashcode was null then icon won't load up. In general file types, hashcode is not utilized anywhere so nothing will happen.
I'll better document this.

Reply:
`readInt ()` [returns primitive int](https://developer.android.com/reference/android/os/Parcel#readint), can't return null.

Reply:
I've put in a fix in another commit.